### ISSUE TITLE
Redeclaration of block scoped variables in the same scope should throw TypeError

### ISSUE THREAD

**Issue by gaearon** [6to5/6to5#557]
Example [from MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let):

```js
switch (x) {
  case 0:
    let foo;
    break;
    
  case 1:
    let foo; // TypeError for redeclaration.
    break;
}
```

6to5 compiles this to 

```js
"use strict";

switch (x) {
  case 0:
    var foo = undefined;
    break;

  case 1:
    var foo = undefined;
    break;
}
```

and doesn't throw.

**Comment by gaearon** [comment]
Also I think @alexkuz is right.
While above example shouldn't work, it *should* work if we wrap each case in a block of its own:

```js
switch (x) {
  case 0:
    { let foo; }
    break;

  case 1:
    { let foo; } // Compiles!
    break;
}
```

**Comment by sebmck** [comment]
@gaearon @alexkuz's original issue has nothing to do with this. It was about not remapping let variables in different scopes while this is about redefining already declared let variables.

**Comment by alexkuz** [comment]
@sebmck well, at least it's related. My original problem was that when I write code like in the last @gaearon example, wrapping cases in blocks, it compiles (as there is no TypeError thrown), but jshint is not happy with the resulting code. If it compiles with different var names though, it would be ok for jshint.

**Comment by sebmck** [comment]
The only thing that's related is that they both apply to block scoped variables.

**Comment by sebmck** [comment]
Closed via ac373a9e1cd0710b21c1714e9d772a7841cfcae2

**Comment by gaearon** [comment]
@alexkuz 

In this case I believe the problem is in JSHint, rather than 6to5. In a perfect world, we will lint original code with ES6 rules instead of linting transpiler's output. This will soon be possible when ESLint gets ES6 and JSX support (it's in progress).
